### ISSUE TITLE
potential MS2 mz value ambiguity

### ISSUE THREAD

**Issue by christophuv** [OpenMS/OpenMS#6626]
Hello

I am a big fan of TOPPView for the visualization of LCMS datasets. 

Recently, I noticed a problem of mzML files or TOPPView (but I think it is a result of Orbitrap problems that report the isolation windows and precursor mz values in a strange fashion). TOPPView seems to illustrate the precursor mz values of the selectedIon in the mzML files. (e.g. <selectedIon><cvParam cvRef="MS" accession="MS:1000744" name="selected ion m/z" value="447.111659033363")
On the other hand, there is also the isolation window available (e.g., <isolationWindow> <cvParam cvRef="MS" accession="MS:1000827" name="isolation window target m/z" value="449.107238769531"). 

An example is provided below. 

The problem here is that in my data (targeted MSMS mode) only an inclusion for the mass 449.1072 was defined, but at no retention time was any mass close to 447 included in this inclusion list. Moreover, when the MSMS spectra are illustrated, it strangly looks as if the MSMS precursors are suddenly shifted from 449 to 447 since the 447 signals are higher in this area. Only when their intensity drops again, is the 449 precursor correctly reported. 

I think this is a problem of Thermo incorrectly reporting the precursor ion's mz values. 

Do you think it would be good to use the isolation mz value instead for illustration in toppview or provide a method to illustrate either the isolation mz value or the precursor mz value? 

I am looking forward to your ideas on that and am happy to provide further information if you need any. 

Thanks a lot!

Best regards, christoph

![image](https://user-images.githubusercontent.com/78472147/214979666-aa9191c8-8daf-4668-a9d8-f1cccd4721b1.png)

`
<precursorList count="1">
            <precursor spectrumRef="controllerType=0 controllerNumber=1 scan=2004">
              <isolationWindow>
                <cvParam cvRef="MS" accession="MS:1000827" name="isolation window target m/z" value="449.107238769531" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
                <cvParam cvRef="MS" accession="MS:1000828" name="isolation window lower offset" value="0.25" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
                <cvParam cvRef="MS" accession="MS:1000829" name="isolation window upper offset" value="0.25" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
                <userParam name="ms level" value="1"/>
              </isolationWindow>
              <selectedIonList count="1">
                <selectedIon>
                  <cvParam cvRef="MS" accession="MS:1000744" name="selected ion m/z" value="447.111659033363" unitCvRef="MS" unitAccession="MS:1000040" unitName="m/z"/>
                  <cvParam cvRef="MS" accession="MS:1000041" name="charge state" value="1"/>
                  <cvParam cvRef="MS" accession="MS:1000042" name="peak intensity" value="2.138346515625e06" unitCvRef="MS" unitAccession="MS:1000131" unitName="number of detector counts"/>
                </selectedIon>
              </selectedIonList>
              <activation>
                <cvParam cvRef="MS" accession="MS:1000422" name="beam-type collision-induced dissociation" value=""/>
                <cvParam cvRef="MS" accession="MS:1000045" name="collision energy" value="45.0" unitCvRef="UO" unitAccession="UO:0000266" unitName="electronvolt"/>
              </activation>
            </precursor>
          </precursorList>
`

**Comment by jpfeuffer** [comment]
I agree it would be nice to have the option to choose which value to display (maybe even both with different symbols if enabled). But I can't say when we will find time for that.
How would it impact your work? Why do you prefer having the center of the isolation window instead of the actual chosen precursor ion? Because the annotation of the actual precursor is a bit arbitrary or even wrong sometimes?

**Comment by christophuv** [comment]
Hello
Thank you for your quick response and the link to the other thread/issue. 

Currently, it is a mere illustration issue for me. 
In my case (the above example) the 447 mz was never put onto the inclusion list used for generating the MSMS dataset. Also, when I look at the specified isolation window, it states 449 +- 0.25, so the precursor is nowhere near this window. 
Moreover, the MSMS spectra of the "precursors" 447@794sec and 449@827sec almost look identically with the principal ions (449, 329, 299) being the ones from the mz 449 ion. Also, in this case the mz difference between 447 and 449 is 1.9949, so not just two hydrogen atoms that could somehow explain the quite similar MSMS patterns (open/closed double bond). 

Furthermore, while I am primarily using OpenMS for the visualization of the data with TOPPView and I not too familiar with the processing pipelines, I can imagine that some of the tools might also use this precursor information and then e.g. report incorrect MSMS spectra for a feature. In this case 447 would incorrectly be annotated with the MSMS spectra of 449

mz 449 @ 794 sec
![449](https://user-images.githubusercontent.com/78472147/215457280-5f13b9f1-96b8-4ec2-8294-ef3f335b2859.png)

mz 447 @ 827 sec
![447](https://user-images.githubusercontent.com/78472147/215457352-b5168933-3d63-42d4-9d6d-a0ca3b5956f3.png)

Thus, I think the precursor MZ value is simply wrong, wherever this issue might originate from. 

best regards, christoph
